adlfs/spec.py

```diff
@@ -1269,6 +1272,32 @@ def cat(self, path, recursive=False, on_error="raise", **kwargs):
         else:
             return self.cat_file(paths[0])
 
+    def url(self, path, expires=3600, **kwargs):
+        """Generate presigned URL to access path by HTTP
+
+        Parameters
+        ----------
+        path : string
+            the key path we are interested in
+        expires : int
+            the number of seconds this signature will be good for.
+        """
+        container_name, blob = self.split_path(path)
+
+        sas_token = generate_blob_sas(
+            account_name=self.account_name,
+            container_name=container_name,
+            blob_name=blob,
+            account_key=self.account_key,
+            permission=BlobSasPermissions(read=True),
+            expiry=datetime.utcnow() + timedelta(seconds=expires),
+        )
+
+        cc = self.service_client.get_container_client(container_name)
```

Comment:
Please convert this to async, as:
```
async with self.service_client.get_blob_client(container_name, blob) as bc:
    url = BlobClient.from_blob_url(bc.url, credential=XXX).url

return url
```

This will help avoid hanging connections.  Would also prefer to see these imports moved inside the method call.

Reply:
I was thinking about this a bit more.  I'm not sure you need the `BlobClient.from_blob_url(xyz...).url`

Shouldn't the following work?
```
async with self.service_client.get_blob_client(container_name, blob) as bc:
    url = bc.from_blob_url(bc.url, credential=XXX).url
return url
```

Reply:
I have incorporated this change.